app/Jobs/Warnings/WarningFetcher.php

```diff
@@ -70,62 +99,46 @@ private function fetchIpmaWarnings(): bool
-            $awarenessIsAllowed = \in_array($awarenessLevelId, self::ALLOWED_AWARENESS_LEVELS, true);
-            if (empty($awarenessLevelId) || ! $awarenessIsAllowed) {
-                continue;
-            }
-
-            DB::transaction(function () use ($data, $awarenessLevelId) {
-                $startedAt = $this->carbonise($data['startTime'])->toDateTimeString();
-                $endedAt = $this->carbonise($data['endTime'])->toDateTimeString();
-
-                $warningExists = IpmaWarning::query()
-                    ->where('area_id', $data['idAreaAviso'])
-                    ->where('awareness_level_id', $awarenessLevelId)
-                    ->where('is_active', 1)
-                    ->where('started_at', '<=', $startedAt)
-                    ->where('ended_at', '>', $startedAt)
-                    ->where('started_at', '<', $endedAt)
-                    ->where('ended_at', '>=', $endedAt)
-                    ->first();
-
-                if (! $warningExists) {
-                    $areasMap = config('ipma.areas', []);
-
-                    $warning = new IpmaWarning();
-                    $warning->uuid = Uuid::uuid4();
-                    $warning->text = ! empty($data['text']) ? $data['text'] : null;
-                    $warning->awareness_type_name = $data['awarenessTypeName'];
-                    $warning->zone_name = Arr::get($areasMap, $data['idAreaAviso'].'.zone');
-                    $warning->area_id = $data['idAreaAviso'];
-                    $warning->area_name = Arr::get($areasMap, $data['idAreaAviso'].'.area');
-                    $warning->awareness_level_id = $awarenessLevelId;
-                    $warning->started_at = $startedAt;
-                    $warning->ended_at = $endedAt;
-                    $warning->save();
-
-                    $this->logger->info(\sprintf(
-                        'Created Ipma warning %s',
-                        $warning->id
-                    ));
+        $ipmaWarnings = collect($response['data'])
+            ->filter(function ($value) {
+                if (! empty($value['awarenessLevelID'])) {
+                    $awarenessLevelId = \mb_strtolower($value['awarenessLevelID']);
+                    $awarenessIsAllowed = \in_array($awarenessLevelId, self::ALLOWED_AWARENESS_LEVELS, true);
+                    if ($awarenessIsAllowed) {
+                        return $value;
+                    }
                 }
+            })->map(function ($item) {
+                // get aux data
+                $countyId = Arr::get(self::AREA_CODES_MAP, $item['idAreaAviso']);
+                $county = County::find($countyId);
+                $district = $county->district;
+                $region = ! \in_array($district->id, [19, 20], true) ? 'CONTINENTE' : $district->name;
+
+                // set/modify attributtes
+                $item['id'] = Uuid::uuid4()->toString();
+                $item['region'] = $region;
```

Comment:
Get rid of the ` $item['region']` and ` $item['county']`, in favour of having the relation in the JSON API spec response.

Reply:
Keep in mind the data here its not a model and i can't eager load the relations.